src/syscall.c

```diff
@@ -6,9 +6,11 @@
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <time.h>
+#include <sys/time.h>
 
+#include "riscv_private.h"
```

Comment:
Can you avoid include "riscv_private.h"? This header was intentionally include by the execution engine itself.